goal_src/jak2/levels/common/airlock.gc

```diff
@@ -7,3 +7,1762 @@
+    (com-airlock-method-27 (_type_ symbol) none :behavior com-airlock 27)
+    )
+  )
+
+
+(defmethod deactivate com-airlock ((obj com-airlock))
+  (process-entity-status! obj (entity-perm-status subtask-complete) #f)
+  (if (nonzero? (-> obj sound-id))
+      (sound-stop (-> obj sound-id))
+      )
+  (if (nonzero? (-> obj gear-sound-id))
+      (sound-stop (-> obj gear-sound-id))
+      )
+  ((method-of-type process-drawable deactivate) obj)
+  (none)
+  )
+
+;; WARN: Return type mismatch process-drawable vs com-airlock.
+(defmethod relocate com-airlock ((obj com-airlock) (arg0 int))
+  (if (nonzero? (-> obj gear))
+      (&+! (-> obj gear) arg0)
+      )
+  (the-as com-airlock ((method-of-type process-drawable relocate) obj arg0))
+  )
+
+(defmethod init-airlock! com-airlock ((obj com-airlock))
+  (process-entity-status! obj (entity-perm-status subtask-complete) #f)
+  (process-drawable-from-entity! obj (the-as entity-actor (-> obj entity)))
+  (logclear! (-> obj mask) (process-mask actor-pause))
+  (set! (-> obj were-behind?) #f)
+  (set! (-> obj inner?)
+        (logtest? (the-as int (res-lump-value (-> obj entity) 'options uint128 :time -1000000000.0)) 1)
+        )
+  (set! (-> obj sound-behind?) #f)
+  (set! (-> obj saw-pilot?) (the-as handle #f))
+  (set! (-> obj open-frame) 0.0)
+  (set! (-> obj pre-open-frame) 0.0)
+  (set! (-> obj lock-frame) 0.0)
+  (set! (-> obj pre-open-speed) 2.0)
+  (set! (-> obj open-distance) (res-lump-float (-> obj entity) 'distance :default 143360.0))
+  (set! (-> obj active-distance)
+        (res-lump-float (-> obj entity) 'idle-distance :default (+ 143360.0 (-> obj open-distance)))
+        )
+  (set! (-> obj y-height) (res-lump-data (-> obj entity) 'height vector))
+  (set! (-> obj level-name) (res-lump-struct (-> obj entity) 'on-notice string))
+  (set! (-> obj open-test)
+        (the-as symbol ((method-of-type res-lump get-property-struct)
+                        (-> obj entity)
+                        'open-test
+                        'interp
+                        -1000000000.0
+                        (the-as structure '(not (or (scene-player?) (focus-test? *target* grabbed))))
+                        (the-as (pointer res-tag) #f)
+                        *res-static-buf*
+                        )
+                )
+        )
+  (set! (-> obj sound-gear) #f)
+  (set! (-> obj sound-pre-open) #f)
+  (set! (-> obj sound-pre-open-stop) #f)
+  (set! (-> obj sound-lock-loop) #f)
+  (set! (-> obj sound-lock-stop) #f)
+  (set! (-> obj sound-post-close) #f)
+  (set! (-> obj sound-post-close-stop) #f)
+  (set! (-> obj sound-open) #f)
+  (set! (-> obj sound-close) #f)
+  (set! (-> obj sound-open-loop) #f)
+  (set! (-> obj sound-close-loop) #f)
+  (set! (-> obj sound-open-stop) #f)
+  (set! (-> obj sound-close-stop) #f)
+  (set! (-> obj door-radius) 20480.0)
+  obj
+  )
+
+(defmethod com-airlock-method-25 com-airlock ((obj com-airlock) (arg0 vector) (arg1 symbol))
```

Comment:
It looks like this function checks the distance from the airlock, and runs on the `on-cross` script if you cross through it. Maybe this could be called `check-crossing-distance`